mkdocs_rss_plugin/util.py

```diff
@@ -593,9 +594,51 @@ def filter_pages(pages: list, attribute: str, length: int) -> list:
                     "guid": page.guid,
                     "image": page.image,
                     "link": page.url_full,
-                    "pubDate": format_datetime(dt=getattr(page, attribute)),
+                    "pubDate": format_datetime(dt=pub_date),
+                    "pubDate3339": pub_date.isoformat("T"),
                     "title": page.title,
                 }
             )
 
         return filtered_pages
+
+    @staticmethod
+    def feed_to_json(feed: dict, *, updated: bool = False) -> dict:
+        """Format internal feed representation as a JSON Feed compliant dict.
+
+        :param feed: internal feed structure, i. e.
+            GitRssPlugin.feed_created/feed_updated value
+        :type feed: dict
+        :param updated: True if this is a feed_updated
+        :type updated: bool
+
+        :return: dict that can be passed to json.dump
+        :rtype: dict
+        """
+        entry_date_key = "date_modified" if updated else "date_published"
+
+        return {
+            "version": "https://jsonfeed.org/version/1",
+            "title": feed.get("title"),
+            "home_page_url": feed.get("html_url"),
+            "feed_url": feed.get("json_url"),
+            "description": feed.get("description"),
+            "icon": feed.get("logo_url"),
+            "authors": [{"name": feed.get("author")}]
+            if feed.get("author") is not None
+            else [],
+            "language": str(feed.get("language")),
+            "items": [
+                {
+                    "id": item.get("guid"),
+                    "url": item.get("link"),
+                    "title": item.get("title"),
+                    "content_html": item.get("description"),
+                    "image": (item.get("image") or (None,))[0],
+                    entry_date_key: item.get("pubDate3339"),
+                    "authors": [{"name": name} for name in (item.get("authors") or ())],
+                    "tags": item.get("categories"),
+                }
+                for item in feed.get("entries", ())
+            ],
+        }
```

Comment:
I was a bit hesitant as to whether put this in a helper method like this or a template like this:

```twig
{{ {
  "version": "https://jsonfeed.org/version/1",
  "title": feed.title,
} | tojson }}
```

The template approach feels more in line with the current system, but on the other hand spawning Jinja just to make one JSON feels a bit weird. @Guts What do you think?

Reply:
I get your dilemma. Let's move on with this implementation using JSON module and when a fully featured customization mechanism will be needed, we'll reconsider that.